File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:

    int maxProfit(vector<int>& prices) {

        int min_till_now=INT_MAX;
        int ans=0;
        for(int i=0;i<prices.size();i++)
        {
            min_till_now=min(min_till_now,prices[i]);
            ans=max(ans,prices[i]-min_till_now);
        }
        return ans;

    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var temp=0;
    var n=prices.length
    var min_till_now=1e9;
    for(var i=0;i<n;i++)
    {
        temp=Math.max(temp,prices[i]-min_till_now);
        min_till_now=Math.min(min_till_now,prices[i]);
    }

    return temp
};